nav2_costmap_2d/include/nav2_costmap_2d/collision_base.hpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2020 Shivang Patel
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_COSTMAP_2D__COLLISION_BASE_HPP_
+#define NAV2_COSTMAP_2D__COLLISION_BASE_HPP_
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <algorithm>
+
+#include "rclcpp/rclcpp.hpp"
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "geometry_msgs/msg/pose2_d.hpp"
+#include "nav2_costmap_2d/costmap_2d.hpp"
+#include "nav2_costmap_2d/costmap_subscriber.hpp"
+#include "nav2_costmap_2d/footprint_subscriber.hpp"
+#include "nav2_util/robot_utils.hpp"
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpedantic"
+#include "tf2/utils.h"
+#pragma GCC diagnostic pop
+
+namespace nav2_costmap_2d
+{
+typedef std::vector<geometry_msgs::msg::Point> Footprint;
+
+class CollisionBase
```

Comment:
I think this should be called `FootprintCollisionChecker` and we should rename the collision checker using subscriptions as something else. This is the "core" collision checking capability and that's just s ROS wrapper for it.

Reply:
You want to call it `CollisionCheckerSub`?

Reply:
Call what? I think this comment is for the core collision stuff. If you're talking about the subscription based one in another file, I think `ExternalCollisionChecker` perhaps? I'm also not great with naming. Something that gets it across its for use outside of the costmaps where you need to subscribe to a costmap topic

Reply:
I am fine with `ExternalCollisionChecker`!

Reply:
@shivaang12 `CostmapTopicCollisionChecker` lets just go full on wordy.